File: source/backend-vacatures-bron/src/main/java/com/cap/pocvng/entity/Straatadres.java
package com.cap.pocvng.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.validation.Valid;

/**
 * Straatadres
 */

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Straatadres extends AdresDetailsNederland {

    @Valid
    @Embedded
    private StraatadresImpl straatadres;

}

File: source/frontend/src/app/entities/vacature/straatadres.ts
import {StraatadresImpl} from './straatadresImpl';

export interface Straatadres {
  straatadres?: StraatadresImpl;
}

export function isStraatadres(object: any): object is Straatadres {
  return 'straatadres' in object;
}
